tests/test_cli.py

```diff
@@ -0,0 +1,174 @@
+    :return: None
+    """
+    base_one = ["./grinder.py", "-r"]
+    flags = ["-r", "-u", "-cu", "-cp", "-ni"]
+    invalid_value = "wrong_one"
+    for f in flags:
+        base_one.extend([f, invalid_value])
+        subprocess.run(base_one)
+        out, err = capfd.readouterr()
+        assert "unrecognized arguments:" in str(err)
+        base_one.pop()
+        subprocess.run(base_one)
+        out, err = capfd.readouterr()
+        assert (
+            "Oops! File with queries was not found. Create it or set name properly."
+            in str(out)
+        )
+        base_one.pop()
+
+
+def test_cli_invalid_args(cli_args: list, capfd) -> None:
+    """
+    Check behavior in case of invalid args with different flags
+    :param cli_args: base arguments
+    :return: None
+    """
+    base_args = cli_args
+    invalid_value = "wrong_one"
+    invalid_args = base_args
+
+    invalid_args.append(invalid_value)
+    subprocess.run(invalid_args)
+    out, err = capfd.readouterr()
+    assert "unrecognized arguments:" in str(err)
+
+    invalid_args = base_args
+    invalid_args.extend(["-nm", "-vs", "-sc", "-vs", "-ts"])
+    flags = ["-cm", "-sm", "-nw", "-vw", "-ht", "-tp", "-ml", "-q", "-v", "-tsp", "-vk"]
+
+    expected_invalid_errors = [
+        "argument -cm/--censys-max: invalid int value",
+        "argument -sm/--shodan-max: invalid int value",
+        "argument -nw/--nmap-workers: invalid int value",
+        "argument -vw/--vulners-workers: invalid int value",
+        "argument -ht/--host-timeout: invalid int value",
+        "argument -tp/--top-ports: invalid int value",
+        "argument -ml/--max-limit: invalid int value",
+        "unrecognized arguments:",
+        "unrecognized arguments:",
+        "unrecognized arguments:",
+        "unrecognized arguments:",
+    ]
+
+    for i in range(len(flags)):
+        invalid_args.extend([flags[i], invalid_value])
+        subprocess.run(invalid_args)
+        err1 = capfd.readouterr().err
+        assert expected_invalid_errors[i] in str(err1)
+
+        invalid_args.pop()
+        subprocess.run(invalid_args)
+        out, err = capfd.readouterr()
+        if str(err) != str(err1):
+            assert "expected one argument" in str(err)
+        invalid_args.pop()
+
+
+def test_cli_vendor_confidence(cli_args: list, query_random: str, capfd) -> None:
+    """
+    Check behavior in case of different vendor confidence
+    :param cli_args: base arguments
+    :return: None
+    """
+    base_args = cli_args
+    query = query_random
+    base_args.extend(["-q", query, "-vc"])
+    vendor_confidence = ["firm", "tentative", "wrong_one"]
+    error = "Confidence level for vendors is not valid"
+
+    for vendor in vendor_confidence:
+        base_args.append(vendor)
+        subprocess.run(base_args)
+        out, err = capfd.readouterr()
+        if error in str(out):
+            assert error in str(out)
```

Comment:
Simply, you do this:
```python
In [50]: out = "test this string"

In [51]: if "test" in out:
    ...:     assert "test" in out
    ...:
```
`AssertionError` will never happen, because you suppress `assert` with `if` clause - if `"test"` in `out` variable and you already check it with `if`, why you try to assert it? This assert will always be `True`. You need to remove this `if` clause to make this test work as expected.

Reply:
Also, I think that you need to check these confidence levels separately:
```python
valid_vendor_confidence = ["certain", "firm", "tentative"]
invalid_vendor_confidence = ["invalid", "vendor", "confidence"]
```